### PR TITLE
update docs to RancherOS from old Microkernel

### PR DESCRIPTION
**Pull request by lanchongyizu**
**Background**

We need update docs to RancherOS from old Microkernel. It's related with https://rackhd.atlassian.net/browse/RAC-6580.

**Done**

* Update microkernel section
* Update raid configuration
* Update debug guide.

@RackHD/corecommitters @pengz1 @nortonluo

### CODE REVIEW

File: docs/devguide/debugging_guide.rst [hunk 1]
```diff
@@ -36,44 +36,70 @@ Default discovery workflow
     title Default Discovery Workflow
     Server->RackHD: DHCP from PXE(nic or BIOS)
     RackHD->Server: ISC DHCP response with IP
+    note over RackHD:
+        If the node is already "known",
```
Comment (pengz1) [inline docs/devguide/debugging_guide.rst]: Known is confusing, we should use discovered.

File: docs/devguide/debugging_guide.rst [hunk 2]
```diff
@@ -36,44 +36,70 @@ Default discovery workflow
...
-    note right of RackHD:
-        multiline
+    note over RackHD:
         RackHD looks up IP address of HTTP request from iPXE script to find the node via its mac-address.
-        If the node is already "known", it will only respond if there's an active workflow that's been invoked related to the node
-        If the node isn't known, it will create a workflow (default is the workflow 'Graph.Sku.Discovery') and respond with an iPXE script to initiate that
-        end note
+        1) If the node is already "known", it will only respond if there's an active workflow
+        that's been invoked related to the node.
+        2) If the node isn't known, it will create a workflow (default is the workflow 'Graph.Sku.Discovery')
```
Comment (pengz1) [inline docs/devguide/debugging_guide.rst]: Known is confusing, suggest 'discovered'

File: docs/devguide/debugging_guide.rst [hunk 3]
```diff
@@ -36,44 +36,70 @@ Default discovery workflow
...
-        the discovery workflow will also create IPMI pollers for the node if relevent information can be found in the catalog
-        end note
+    note over RackHD
+        Task output stored as catalogs in RackHD related to the node.
+        If RackHD is configured with SKU definitions,
+        it processes these catalogs to determine the SKU.
+        If there's a SKU specific workflow defined, control is continued to that.
+        The discovery workflow will create an enclosure node based on the catalog data.
+        The discovery workflow will also create IPMI pollers for the node,
+        if relevent information can be found in the catalog.
```
Comment (pengz1) [inline docs/devguide/debugging_guide.rst]: There is also a task to generate tag.

File: docs/devguide/naming_conventions.rst [hunk 4]
```diff
@@ -31,26 +31,26 @@ Examples::
...
+tar archives contain docker image.
 
 Examples::
 
-    overlayfs_3.13.0-32_flashupdt.cpio.gz
-    overlayfs_3.13.0-32_brocade.cpio.gz
-    overlayfs_3.13.0-32_all_binaries.cpio.gz
+    micro_1.2.0_flashupdt.docker.tar.xz
+    micro_1.2.0_brocade.docker.tar.xz
+    micro_1.2.0_all_binaries.docker.tar.xz
```
Comment (pengz1) [inline docs/devguide/naming_conventions.rst]: This is not the case till now.

File: docs/rackhd/configuration.rst [hunk 5]
```diff
@@ -514,56 +514,48 @@ The taskgraph endpoint is the interface that is used by nodes to interacting wit
...
-.. code-block:: shell
-
-   sudo pip uninstall ansible
-   sudo dpkg -r
-   sudo apt-get purge ansible
-   sudo pip –U install pip
+(2). Refer to the Requirements section of the Readme in the on-imagebuilder repo to install latest version of docker: https://github.com/RackHD/on-imagebuilder#requirements
 
-(3). Refer to the Requirements section of the Readme in the on-imagebuilder repo to install latest version of ansible: https://github.com/RackHD/on-imagebuilder#requirements
+(3). For Quanta storcli - https://github.com/RackHD/on-imagebuilder#oem-tools
```
Comment (pengz1) [inline docs/rackhd/configuration.rst]: Storcli is not specified for quanta, it is common tool for LSI Megaraid series.

File: docs/devguide/debugging_guide.rst (hunk 2 above)
Comment (nortonluo) [reply docs/devguide/debugging_guide.rst]: How about ""Default workflow is 'Graph.Sku.Discovery'"?

File: docs/devguide/debugging_guide.rst (hunk 1 above)
Comment (lanchongyizu) [reply docs/devguide/debugging_guide.rst]: @pengz1 A new node can be created manually with the API `POST /api/2.0/nodes`. So I think the original expression `known` is more suitable than `discovered` which only refers to the nodes that are discovered but not manually created.

File: docs/devguide/debugging_guide.rst [hunk 6]
```diff
@@ -36,44 +36,70 @@ Default discovery workflow
...
+    end note
     Server->RackHD: IPXE script requests what to do from RackHD (http)
-    note right of RackHD:
-        multiline
+    note over RackHD:
         RackHD looks up IP address of HTTP request from iPXE script to find the node via its mac-address.
-        If the node is already "known", it will only respond if there's an active workflow that's been invoked related to the node
-        If the node isn't known, it will create a workflow (default is the workflow 'Graph.Sku.Discovery') and respond with an iPXE script to initiate that
-        end note
+        1) If the node is already "known", it will only respond if there's an active workflow
```
[the remark this answers is not in the document]
Comment (lanchongyizu) [reply docs/devguide/debugging_guide.rst]: See the above.

File: docs/devguide/naming_conventions.rst (hunk 4 above)
Comment (lanchongyizu) [reply docs/devguide/naming_conventions.rst]: This is a name convention that is better for user to have. It is not required for `on-imagebuilder`.

File: docs/devguide/debugging_guide.rst (hunk 2 above)
Comment (lanchongyizu) [reply docs/devguide/debugging_guide.rst]: See the above.

File: docs/devguide/naming_conventions.rst (hunk 4 above)
Comment (pengz1) [reply docs/devguide/naming_conventions.rst]: If even we do not use it, why should we propose it?

File: docs/devguide/debugging_guide.rst (hunk 3 above)
Comment (lanchongyizu) [reply docs/devguide/debugging_guide.rst]: Done

**Pull request merged**